lenthall/src/main/scala/lenthall/collections/EnhancedCollections.scala

```diff
@@ -25,5 +27,50 @@ object EnhancedCollections {
 
     def firstByType[T <: T2](implicit tag: ClassTag[T]): Option[T] = traversable collectFirst { case t: T => t }
   }
+  
+  implicit class EnhancedTable[R, C, V](val table: Table[R, C, V]) extends AnyVal {
```

Comment:
yeah we could, I'm generally not a mega fan of wrapping single values in case classes but I know we love our types :p 
For the record there's a guava implementation of this (but in Java so less easy to manipulate), and I found a scala one but based on `Vector`s but we really want hash maps to speed up lookups